### PR TITLE
[release-1.14] update prow to 1.24 and fix remote-secret creation

### PR DESCRIPTION
**Pull request by stevenctl**
* prow: update default image to 1.24.0
* wait for token data to be available
* fix lint and release note
* unit test coverage

### CODE REVIEW

**Review by lei-tang — APPROVED** [review]

File: istioctl/pkg/multicluster/remote_secret.go [hunk 1]
```diff
@@ -255,36 +291,38 @@ func getOrCreateServiceAccountSecret(
...
 		}
 	}
 
 	// finally, create the sa token secret manually
 	// https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-service-account-api-token
 	// TODO ephemeral time-based tokens are preferred; we should re-think this
-	return client.CoreV1().Secrets(opt.Namespace).Create(ctx, &v1.Secret{
+	log.Infof("Creating token secret for service account %q", serviceAccount.Name)
+	secretName := tokenSecretName(serviceAccount.Name)
+	return client.Kube().CoreV1().Secrets(opt.Namespace).Create(ctx, &v1.Secret{
```
Comment (GregHanson) [inline istioctl/pkg/multicluster/remote_secret.go:323]: @stevenctl does this `client.CoreV1().Secrets` -> `client.Kube().CoreV1().Secrets` work as expected without the underlying change to `pkg/kube/client.go` from the original PR?
https://github.com/istio/istio/pull/38789/files#diff-137bba79988de442c1fde0fc29520d6d7e5a5239ceac82f5439a5bf513ebb2d4

**Review by GregHanson — CHANGES_REQUESTED** [review]

**Comment by GregHanson** [comment]
resolves https://github.com/istio/istio/issues/38987

File: istioctl/pkg/multicluster/remote_secret.go (hunk 1 above)
Comment (stevenctl) [reply istioctl/pkg/multicluster/remote_secret.go:323]: Nope, I think https://github.com/istio/istio/pull/38789/files#diff-137bba79988de442c1fde0fc29520d6d7e5a5239ceac82f5439a5bf513ebb2d4L111

Line 111 in the "before" part of the diff already had the `Kube()` method returning the embedded client.

Comment (stevenctl) [reply istioctl/pkg/multicluster/remote_secret.go:323]: So no behavior change.

**Review by GregHanson — APPROVED** [review]

**Pull request merged**